conformance/lang/expressions/multiplicative-expr/division.balt

```diff
@@ -4033,3 +4033,851 @@ Description: Test integer division panic at module-level.
+
+function init() {
+    decimal a = -0.0d;
+    decimal b = 0.0d;
+    decimal c = a / b; // @panic decimal operation resulting in unsupported decimal value 'NaN'
+}
+
+Test-Case: output
+Description: Test division scenarios for different decimal values.
+Labels: multiplicative-expr, DecimalFloatingPointNumber, decimal, DecimalTypeSuffix, unary-minus
+
+function init() {
+    decimal a = 2.5E-17d;
+    decimal b = 2.5E+17d;
+    decimal c = b / a;
+    io:println(c); // @output 1E+34
+    io:println(2.5E+17d / 2.5E-17d); // @output 1E+34
+
+    a = 1.423223E6d;
+    b = 2.34413E2d;
+    c = a / b;
+    io:println(c); // @output 6071.433751541083472333019073174269
+    io:println(1.423223E6d / 2.34413E2d); // @output 6071.433751541083472333019073174269
+
+    a = 25.0E1742d;
+    b = 1.0d;
+    c = a / b;
+    io:println(c); // @output 2.5E+1743
+    io:println(25.0E1742d / 1.0d); // @output 2.5E+1743
+
+    a = 25.0E17d;
+    b = 0.9E-10d;
+    c = a / b;
+    io:println(c); // @output 27777777777777777777777777777.77778
+    io:println(25.0E17d / 0.9E-10d); // @output 27777777777777777777777777777.77778
+
+    a = -25.0E17d;
+    b = 0.9E-10d;
+    c = a / b;
+    io:println(c); // @output -27777777777777777777777777777.77778
+    io:println(-25.0E17d / 0.9E-10d); // @output -27777777777777777777777777777.77778
+
+    a = 17.0E1290d;
+    b = 13.0E1521d;
+    c = a / b;
+    io:println(c); // @output 1.307692307692307692307692307692308E-231
+    io:println(17.0E1290d / 13.0E1521d); // @output 1.307692307692307692307692307692308E-231
+}
+
+Test-Case: output
+Description: Test division scenarios for different nillable decimal values.
+Labels: multiplicative-expr, DecimalFloatingPointNumber, decimal, DecimalTypeSuffix, optional-type, numeric-nil-lifting, unary-minus, nil-literal, value:toBalString
+
+function init() {
+    decimal? a = 2.5E-17d;
+    decimal? b = 2.5E+17d;
+    decimal? c = b / a;
+    io:println(c); // @output 1E+34
+
+    a = 1.423223E6d;
+    b = 2.34413E2d;
+    c = a / b;
+    io:println(c); // @output 6071.433751541083472333019073174269
+
+    a = 25.0E1742d;
+    b = 1.0d;
+    c = a / b;
+    io:println(c); // @output 2.5E+1743
+
+    a = 25.0E17d;
+    b = 0.9E-10d;
+    c = a / b;
+    io:println(c); // @output 27777777777777777777777777777.77778
+
+    a = -25.0E17d;
+    b = 0.9E-10d;
+    c = a / b;
+    io:println(c); // @output -27777777777777777777777777777.77778
+
+    a = 17.0E1290d;
+    b = 13.0E1521d;
+    c = a / b;
+    io:println(c); // @output 1.307692307692307692307692307692308E-231
+
+    a = 17.0E1290d;
+    b = ();
+    c = a / b;
+    io:println(c.toBalString()); // @output ()
+
+    a = ();
+    b = 13.0E1521d;
+    c = a / b;
+    io:println(c.toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test division scenarios for different float values.
+Labels: multiplicative-expr, DecimalFloatingPointNumber, float, FloatTypeSuffix, unary-minus
```

Comment:
Need to use `FloatingPointTypeSuffix` instead of `FloatTypeSuffix`. Should fix in other places too.